zipline/gens/tradesimulation.py

```diff
@@ -296,7 +298,9 @@ def _get_daily_message(self, dt, algo, perf_tracker):
         Get a perf message for the given datetime.
         """
         perf_message = perf_tracker.handle_market_close(
-            dt, self.data_portal,
+            dt=dt,
+            data_portal=self.data_portal,
+            calculate_position_weights=algo.calculate_expected_shortfall,
```

Comment:
This feels like a poor design to me but I'm not sure a better way to go about it.